src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -1,653 +1,657 @@
+				is_skip_cur_tcm = true;
+			}
+			else
+			{
+				//update the number of times a TCM being broadcast within time out period
+				times += 1; //Increase by 1 for every iteration
+				itr->second.num_of_times = times;
+				PLOG(logDEBUG) << " TCMs reqid = " << tcmv01_req_id_hex<< " repeatedly broadcast " << times  << " times." << std::endl;
+			}
+		}
+	}	
+
+	if(!is_tcm_hex_found)
+	{
+		//Initialize the number of times a TCM being broadcast
+		TCMBroadcastMetadata t;
+		t.num_of_times = 1;
+		t.tcm_hex = tcm_hex_payload;
+		_tcm_broadcast_times->insert( {tcmv01_req_id_hex, t});
+		PLOG(logDEBUG) << " TCMs reqid =  "<< tcmv01_req_id_hex << " has been broadcast once."<< std::endl;
+	} 
+	return is_skip_cur_tcm;
+}
+
+int CARMACloudPlugin::StartWebService()
+{
+	//Web services 
+	char *placeholderX[1]={0};
+	int placeholderC=1;
+	QCoreApplication a(placeholderC,placeholderX);
+
+ 	QHostAddress address = QHostAddress(QString::fromStdString (webip));
+	quint16 port = static_cast<quint16>(webport);
+
+	QSharedPointer<OpenAPI::OAIApiRequestHandler> handler(new OpenAPI::OAIApiRequestHandler());
+	handler = QSharedPointer<OpenAPI::OAIApiRequestHandler> (new OpenAPI::OAIApiRequestHandler());
+
+	auto router = QSharedPointer<OpenAPI::OAIApiRouter>::create();
+    router->setUpRoutes();
+
+    QObject::connect(handler.data(), &OpenAPI::OAIApiRequestHandler::requestReceived, [&](QHttpEngine::Socket *socket) {
+
+		CARMAResponseHandler(socket);
+    });
+
+    QObject::connect(handler.data(), &OpenAPI::OAIApiRequestHandler::requestReceived, [&](QHttpEngine::Socket *socket) {
+
+		router->processRequest(socket);
+    });
+
+    QHttpEngine::Server server(handler.data());
+
+    if (!server.listen(address, port)) {
+        qCritical("Unable to listen on the specified port.");
+        return 1;
+    }
+	PLOG(logINFO)<<"CARMACloudPlugin:: Started web service";
+	return a.exec();
+
+}
+
+
+void CARMACloudPlugin::UpdateConfigSettings() {
+    std::lock_guard<mutex> lock(_cfgLock);
+	GetConfigValue<uint64_t>("Frequency", _frequency);	
+	GetConfigValue<string>("WebServiceIP",webip);
+	GetConfigValue<uint16_t>("WebServicePort",webport);
+	GetConfigValue<uint16_t>("fetchTime",fetchtime);
+	GetConfigValue<string>("MobilityOperationStrategies", _strategies);	
+	GetConfigValue<uint16_t>("TCMRepeatedlyBroadcastTimeOut",_TCMRepeatedlyBroadcastTimeOut);	
+	GetConfigValue<string>("TCMNOAcknowledgementDescription", _TCMNOAcknowledgementDescription);
+	GetConfigValue<int>("TCMRepeatedlyBroadCastTotalTimes", _TCMRepeatedlyBroadCastTotalTimes);
+	GetConfigValue<int>("TCMRepeatedlyBroadcastSleep", _TCMRepeatedlyBroadcastSleep);
+	GetConfigValue<string>("listTCM",list_tcm);
+	std::string carma_cloud_ip;
+	uint carma_cloud_port;
+	GetConfigValue<string>("CARMACloudIP",carma_cloud_ip);
+	GetConfigValue<uint>("CARMACloudPort",carma_cloud_port);
+	carma_cloud_url = carma_cloud_ip + ":" + std::to_string(carma_cloud_port);
```

Comment:
Is there a benefit to having these as separate parameters if we're going to concatenate them anyway? It seems like all it's doing is saving the user from having to know that ":" goes between the domain and the port. Would be simpler I think to just have a "carma_cloud_url" parameter in general?